src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -6435,7 +6458,34 @@ private FlowAnalysisAnnotations GetParameterAnnotations(ParameterSymbol paramete
         {
             // Annotations are ignored when binding an attribute to avoid cycles. (Members used
             // in attributes are error scenarios, so missing warnings should not be important.)
-            return IsAnalyzingAttribute ? FlowAnalysisAnnotations.None : parameter.FlowAnalysisAnnotations;
+            if (IsAnalyzingAttribute)
+                return FlowAnalysisAnnotations.None;
+
+            // Conditional annotations are ignored on parameters of non-boolean members.
+            if (parameter.ContainingSymbol.GetTypeOrReturnType().Type.SpecialType != SpecialType.System_Boolean)
+            {
+                var annotations = parameter.FlowAnalysisAnnotations;
```

Comment:
>var [](http://example.com/codeflow?start=16&length=3)

Consider moving this declaration before the `if` and having a single `return annotations;` at the end of the method.